src/sync/persist-fixup-transformer.test.ts

```diff
@@ -0,0 +1,360 @@
+      },
+      'c/fake00000000000000000000000head2/m': [
+        'fake00000000000000000000000value',
+        'fake000000000000000000000000head',
+      ],
+      'c/fake00000000000000000000000head2/r': 1,
+      'c/fake00000000000000000000000value/d': [0, []],
+      'c/fake00000000000000000000000value/r': 2,
+      'h/test': 'fake00000000000000000000000head2',
+    });
+    assert.equal(newHeadHash2, fakeHash('head2'));
+  }
+});
+
+test('fixup of a single snapshot commit with a btree with internal nodes', async () => {
+  const memdag = new dag.TestStore(
+    undefined,
+    makeNewTempHashFunction(),
+    () => undefined,
+  );
+
+  const entries = Object.entries({
+    a: 0,
+    b: 1,
+    c: 2,
+    d: 3,
+    e: 4,
+    f: 5,
+  });
+
+  const [headHash, valueHash] = await memdag.withWrite(async dagWrite => {
+    const tree = new BTreeWrite(dagWrite, undefined, 2, 4, () => 1, 0);
+    for (const [k, v] of entries) {
+      await tree.put(k, v);
+    }
+
+    const valueHash = await tree.flush();
+    const c = db.newSnapshot(
+      dagWrite.createChunk,
+      null,
+      0,
+      null,
+      valueHash,
+      [],
+    );
+    await dagWrite.putChunk(c.chunk);
+    await dagWrite.setHead('test', c.chunk.hash);
+    await dagWrite.commit();
+    return [c.chunk.hash, valueHash];
+  });
+
+  // console.log(JSON.stringify(memdag.kvStore.snapshot(), null, 2));
+
+  assert.deepEqual(
+    memdag.kvStore.snapshot(),
+
+    {
+      'c/t/000000000000000000000000000000/d': [
+        0,
+        [
+          ['a', 0],
+          ['b', 1],
+        ],
+      ],
+      'c/t/000000000000000000000000000001/d': [
+        0,
+        [
+          ['c', 2],
+          ['d', 3],
+          ['e', 4],
+          ['f', 5],
+        ],
+      ],
+      'c/t/000000000000000000000000000002/d': [
+        1,
+        [
+          ['b', 't/000000000000000000000000000000'],
+          ['f', 't/000000000000000000000000000001'],
+        ],
+      ],
+      'c/t/000000000000000000000000000002/m': [
+        't/000000000000000000000000000000',
+        't/000000000000000000000000000001',
+      ],
+      'c/t/000000000000000000000000000003/d': {
+        meta: {
+          type: 3,
+          basisHash: null,
+          lastMutationID: 0,
+          cookieJSON: null,
+        },
+        valueHash: 't/000000000000000000000000000002',
+        indexes: [],
+      },
+      'c/t/000000000000000000000000000003/m': [
+        't/000000000000000000000000000002',
+      ],
+      'h/test': 't/000000000000000000000000000003',
+      'c/t/000000000000000000000000000000/r': 1,
+      'c/t/000000000000000000000000000001/r': 1,
+      'c/t/000000000000000000000000000002/r': 1,
+      'c/t/000000000000000000000000000003/r': 1,
+    },
+  );
+
+  const mappings = new Map([
+    [headHash, fakeHash('head')],
+    [valueHash, fakeHash('value')],
+    [parseHash('t/000000000000000000000000000000'), fakeHash('data0')],
```

Comment:
Suggests testing the case where mappings does not contain all the temp hashses.  This will happen when the memdag is mutated while we are doing the persist.

Reply:
Good call. This exposed a flaw in the transformer. 

T0:
S1  <- persist on this 

T1:
S1
L1 <- fixup on this still need to transform S1 and its refs